G1ANT.Addon.Net/G1ANT.Addon.Net/Commands/MailImapCommand.cs

```diff
@@ -0,0 +1,137 @@
+﻿/**
+*    Copyright(C) G1ANT Ltd, All rights reserved
+*    Solution G1ANT.Addon, Project G1ANT.Addon.Net
+*    www.g1ant.com
+*
+*    Licensed under the G1ANT license.
+*    See License.txt file in the project root for full license information.
+*
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MailKit;
+using MailKit.Net.Imap;
+using G1ANT.Language;
+using System.Net;
+using MimeKit;
+
+
+namespace G1ANT.Addon.Net
+{
+
+    [Command(Name = "mail.imap", Tooltip = "This command tries to retrieve the mails from inbox.")]
+    public class MailImapCommand : Command
+    {
+        public class Arguments : CommandArguments
+        {
+            [Argument(Required = true, Tooltip = "Host name")]
+            public TextStructure Host { get; set; }
+
+            [Argument(Required = true, Tooltip = "Port")]
+            public IntegerStructure Port { get; set; } = new IntegerStructure(993);
+
+            [Argument(Required = true, Tooltip = "Login of the inbox user")]
+            public TextStructure Login { get; set; }
+
+            [Argument(Required = true, Tooltip = "Password of the inbox user")]
+            public TextStructure Password { get; set; }
+
+            [Argument(Required = true, Tooltip = "Since what date should emails be retrieved")]
+            public DateStructure SinceDate { get; set; }
+
+            [Argument(Required = false, Tooltip = "To what date should emails be retrieved")]
+            public DateStructure ToDate { get; set; } = new DateStructure(DateTime.Now);
+
+            [Argument(Required = false, Tooltip = "Look only for already unread messages")]
+            public BooleanStructure OnlyUnreadMessages { get; set; } = new BooleanStructure(false);
+
+            [Argument(Required = false, Tooltip = "Mark analyzed messages as read")]
+            public BooleanStructure MarkAsRead { get; set; } = new BooleanStructure(true);
+
+            [Argument(Required = false, Tooltip = "Received messages")]
+            public VariableStructure Result { get; set; } = new VariableStructure("result");
+        }
+
+
+        public MailImapCommand(AbstractScripter scripter) : base(scripter)
+        { }
+
+
+        public void Execute(Arguments arguments)
+        {
+            var client = new ImapClient();
+            var credentials = new NetworkCredential(arguments.Login.Value, arguments.Password.Value);
```

Comment:
What will happen if any of the values from the `arguments` (like `MarkAsRead` or `Login`) don't not have a value?

Reply:
MarkAsRead has default value so if it's not provided then it is defaultly set to true:
```
[Argument(Required = false, Tooltip = "Mark analyzed messages as read")]
public BooleanStructure MarkAsRead { get; set; } = new BooleanStructure(true);
```
When the Login value is not provided, then an exception will be thrown. It's due to that it is Required = true in the attributes.
```
[Argument(Required = true, Tooltip = "Login of the inbox user")]
public TextStructure Login { get; set; }
```
https://gyazo.com/696826af75d2e60d2125f7450b693487